lib/KRadioButtonGroup.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+
+  <div class="k-radio-button-group">
+    <!-- @slot To render the KRadioButtons -->
+    <slot></slot>
+  </div>
+
+</template>
+
+
+<script>
+
+  export default {
+    name: 'KRadioButtonGroup',
+    data() {
+      return {
+        radioButtons: [],
+        focusedRadioIdx: 0,
+        firstRadioIdx: 0,
+        lastRadioIdx: 0,
+        radioBtnIdx: 0,
+      };
+    },
+    computed: {
+      isFirefox() {
+        return navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+      },
+    },
+    mounted() {
+      if (!this.isFirefox) {
+        return;
+      }
+      this.$nextTick(() => {
+        const KRadioButtonGroupChildren = this.$children;
+        this.queryAndAddRadioBtns(KRadioButtonGroupChildren);
+
+        this.$el.addEventListener('keyup', this.onKeyUp);
+
+        this.lastRadioIdx = this.radioButtons.length - 1;
+
+        const firstRadioButton = this.radioButtons[this.focusedRadioIdx];
+        firstRadioButton.setTabIndex(0);
+      });
+    },
+
+    methods: {
+      handleInputChange(idx) {
+        if (this.isFirefox) {
+          this.setChecked(idx);
+          this.focusedRadioIdx = idx;
+        }
+      },
+      queryAndAddRadioBtns(children) {
```

Comment:
The recursion never gets to a third child of a childrens array. I know that our motivating use-case has only two grid items so it works there, but the point of recursion here is to make it work no matter what. It'd be best to fix it so it's as robust as possible. You can reproduce by

```
<KGrid>
  <KRadioButtonGroup>
    <KGridItem>
      <KRadioButton
        v-model="exampleValue"
        label="Option A"
        value="val-a"
      />
      <KRadioButton
        v-model="exampleValue"
        label="Option B"
        value="val-b"
      />
    </KGridItem>
    <KGridItem>
      <KRadioButton
        v-model="exampleValue"
        label="Option C"
        description="This one is special!"
        buttonValue="val-c"
      />
      <KRadioButton
        v-model="exampleValue"
        label="Truncated label. Adjusting your browser window size to see this in action."
        buttonValue="val-d"
        truncateLabel
      />
    </KGridItem>
    <KGridItem>
      <KRadioButton
        v-model="exampleValue"
        label="Option E"
        description="This one is special!"
        buttonValue="val-e"
      />
      <KRadioButton
        v-model="exampleValue"
        label="Option F"
        buttonValue="val-f"
        truncateLabel
      />
    </KGridItem>
  </KRadioButtonGroup>
</KGrid>
```

and then see it's not possible to navigate with keyboard to the last two checkboxes:

![Peek 2024-08-16 12-57](https://github.com/user-attachments/assets/3a260c50-6175-4c29-8cc4-286bc8caae19)

Reply:
I think it has to do with the way it is structured - the only thing that makes the function run on a sibling child is `if (children.length > 1) this.queryAndAddRadioBtns(children[1].$children);` but that's not sufficient and can't cover all children.

For recursions, this is a typical structure:

```
FUNCTION traverseChildren(array)
  FOR EACH child IN array
    PROCESS child
    IF child HAS children
      CALL traverseChildren(child.children)
  END FOR
END FUNCTION
```

I think that using it should resolve it.

If that's helpful, one thing that helps to check recursion implementation is to test on crazy unexpected nested structures :) If recursion is implemented correctly it will always work on all children on all levels.

For debugging, this piece may help:
```
queryAndAddRadioBtns(children) {
   console.log('[queryAndAddRadioBtns]', children.map(child => child.$el));
   ...
```

Then just have a look at a browser console and you will see in detail how your recursion progresses on elements and if it covers them all.

Reply:
Minor additional note - since this is a bit more complex piece, would you please add a comment above the function briefly summarizing what it does and why (example of grid usage)?